alerter.c

```diff
@@ -2,13 +2,17 @@
 #include <assert.h>
 
 int alertFailureCount = 0;
-
+void testfail(void);
+    
 int networkAlertStub(float celcius) {
     printf("ALERT: Temperature is %.1f celcius.\n", celcius);
     // Return 200 for ok
     // Return 500 for not-ok
     // stub always succeeds and returns 200
-    return 200;
+    if(celcius<=200)
+        return 200;
+    else 
+        return 500;
 }
 
 void alertInCelcius(float farenheit) {

```

Comment:
The production code should not be aware of test code or test stub. It should be abstracted by the use of interface or function pointer.